src/imputors.jl

```diff
@@ -1,77 +1,129 @@
 """
     Imputor
 
-An imputor stores information about imputing values in `AbstractArray`s and `DataFrame`s.
+An imputor stores information about imputing values in `AbstractArray`s and `Tables.table`s.
 New imputation methods are expected to sutype `Imputor` and, at minimum,
 implement the `impute!{T<:Any}(imp::<MyImputor>, ctx::Context, data::AbstractArray{T, 1})`
 method.
 """
-
 abstract type Imputor end
 
-"""
-    impute!(imp::Imputor, data::Dataset, limit::Float64=0.1)
+# A couple utility methods to avoid messing up var and obs dimensions
+obsdim(imp::Imputor) = imp.vardim == 1 ? 2 : 1
+vardim(imp::Imputor) = imp.vardim
 
-Creates a `Context` using information about `data`. These include
+function obswise(imp::Imputor, data::AbstractMatrix)
+    (imp.vardim == 1 ? view(data, :, i) : view(data, i, :) for i in axes(data, obsdim(imp)))
+end
 
-1. missing data function which defaults to `missing`
+function varwise(imp::Imputor, data::AbstractMatrix)
+    (imp.vardim == 1 ? view(data, i, :) : view(data, :, i) for i in axes(data, vardim(imp)))
+end
 
-2. number of elements: `*(size(data)...)`
+function filterobs(f::Function, imp::Imputor, data::AbstractMatrix)
+    mask = [f(x) for x in obswise(imp, data)]
+    return imp.vardim == 1 ? data[:, mask] : data[mask, :]
+end
```

Comment:
i wonder if there is a faster way, using `filter`

Reply:
Maybe, but I'm inclined to leave that for a performance improvements PR. The internal API would remain the same.

Reply:
https://github.com/invenia/Impute.jl/issues/25